src/GraphQL/Server.hs

```diff
@@ -37,18 +38,16 @@ import GraphQL.Internal.Schema (HasName(..))
 import GraphQL.API
 import GraphQL.Internal.Input (CanonicalQuery)
 
-import Control.Monad.Catch (MonadThrow, throwM, Exception)
+import Control.Monad.Catch (Exception)
+
+data QueryError =
+  QueryError Text
+  | InternalError Text
+  | InvalidFieldError Text
```

Comment:
We talked earlier about having a hard distinction between errors that spring from users (client-side errors, 400s, query errors) and errors that spring from the server code (500s). Which is this?